src/main/java/com/elixr/springbootfile/service/FileServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+package com.elixr.springbootfile.service;
+
+import com.elixr.springbootfile.constants.Constants;
+import com.elixr.springbootfile.dao.ModelFile;
+import com.elixr.springbootfile.exceptionhandling.NotFoundException;
+import com.elixr.springbootfile.repository.FileRepository;
+import com.elixr.springbootfile.response.FileInfoDTO;
+import com.elixr.springbootfile.response.SuccessResponse;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.time.LocalDateTime;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
+public class FileServiceImpl implements FileService {
+
+    private final FileRepository fileRepo;
+
+    @Override
+    public ResponseEntity<?> fileUpload(MultipartFile file, ModelFile modelFile, String userName) {
+        if (Objects.equals(file.getContentType(), "text/plain")) {
+            try {
+                modelFile.setFileName(file.getOriginalFilename());
+                modelFile.setUserName(userName);
+                Files.copy(file.getInputStream(), Paths.get(Constants.UPLOAD_DIR + File.separator, file.getOriginalFilename()), StandardCopyOption.REPLACE_EXISTING);
+                fileRepo.save(modelFile);
+                return new ResponseEntity<>(SuccessResponse.builder().success(Constants.STATUS).id(String.valueOf(modelFile.getId())).build(), HttpStatus.OK);
+            } catch (Exception e) {
+                throw new NotFoundException(Constants.FILE_PATH_NOT_VALID);
```

Comment:
An exception in saving the file should not result in a 404 error but a 500 response to the client

Reply:
I have created a new custom exception for Internal server exception and throwed here.